tests/cpp/nodes/indexing/test_advanced.cpp

```diff
@@ -120,8 +121,6 @@ TEST_CASE("AdvancedIndexingNode") {
                     CHECK(B_ptr->size(state) == s_ptr->size(state));
 
                     CHECK(std::ranges::equal(B_ptr->view(state), std::vector{4}));
-
-                    verify_array_diff({}, {4}, B_ptr->diff(state));
```

Comment:
Don't we need to call propagate on the `ArrayValidationNode` to ensure the diff is valid? Maybe we can switch a lot of these tests to using `graph->propagate()` so we don't have to propagate all the nodes individually...

Reply:
Oh! Good catch. Will fix everywhere.

Reply:
Ok, should be fixed. Also made the tests a bit more future proof by just propagating/commiting/reverting all descendants.